elements/lisk-utils/src/math/q.ts

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable no-bitwise */
+/*
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { MAX_FRAC, ONE, Q_OPERATION, TWO } from './constants';
+
+const numberToQ = (base: bigint, val: number): bigint => {
+	const denominator = TWO ** base;
+	if (Number.isInteger(val)) {
+		return BigInt(val) * denominator;
+	}
+	const [intStr, fractionalStr] = val.toString().split('.');
+	return (
+		BigInt(intStr) * denominator +
+		(BigInt(fractionalStr) * denominator) / BigInt(10) ** BigInt(fractionalStr.length)
```

Comment:
We can completely get rid of the `denominator` variable and replace multiplication with `<< base`.

Reply:
Don't forget the brackets:

```typescript
const numberToQ = (base: bigint, val: number): bigint => {
	if (Number.isInteger(val)) {
		return BigInt(val) << base;
	}
	const [intStr, fractionalStr] = val.toString().split('.');
	return (
		(BigInt(intStr) << base) +
		(BigInt(fractionalStr) << base) / BigInt(10) ** BigInt(fractionalStr.length)
	);
};
```